qiita_ware/test/test_ebi.py

```diff
@@ -136,83 +125,49 @@ def test_add_dict_as_tags_and_values(self):
-                                   self.sample1_fp, 'experiment description',
-                                   'library protocol')
-
-        prep_info = submission.samples['test1']['prep']
-        self.assertEqual(prep_info['platform'], 'ILLUMINA')
-        self.assertEqual(prep_info['file_path'], self.sample1_fp)
-        with self.assertRaises(KeyError):
-            submission.add_sample_prep('test3', 'ILLUMINA', 'fastq',
-                                       self.sample3_fp,
-                                       'experiment description',
-                                       'library protocol')
-
-    def test_add_sample_prep_exception(self):
-        submission = EBISubmission(2, 'ADD')
-        submission.add_sample('test1', '9606', 'homo sapiens', 'desc1')
-        submission.add_sample('test2', '9606', 'homo sapiens', 'desc1')
-        with self.assertRaises(ValueError):
-            submission.add_sample_prep('test2', 'DOES-NOT-EXIST', 'fastq',
-                                       self.sample1_fp,
-                                       'experiment description',
-                                       'library protocol')
-        with self.assertRaises(KeyError):
-            submission.add_sample_prep('test3', 'DOES-NOT-EXIST', 'fastq',
-                                       self.sample3_fp,
-                                       'experiment description',
-                                       'library protocol')
-
-    def test_generate_library_descriptor(self):
-        e = EBISubmission(2, 'ADD')
-        elm = ET.Element('design', {'foo': 'bar'})
+        # removing samples so test text is easier to read
+        keys_to_del = ['1.SKD6.640190', '1.SKM6.640187', '1.SKD9.640182',
+                       '1.SKM8.640201', '1.SKM2.640199', '1.SKD2.640178',
+                       '1.SKB7.640196', '1.SKD4.640185', '1.SKB8.640193',
+                       '1.SKM3.640197', '1.SKD5.640186', '1.SKB1.640202',
+                       '1.SKM1.640183', '1.SKD1.640179', '1.SKD3.640198',
+                       '1.SKB5.640181', '1.SKB4.640189', '1.SKB9.640200',
+                       '1.SKM9.640192', '1.SKD8.640184', '1.SKM5.640177',
+                       '1.SKM7.640188', '1.SKD7.640191', '1.SKB6.640176',
+                       '1.SKM4.640180']
+        for k in keys_to_del:
+            del(submission.samples[k])
+            del(submission.samples_prep[k])
+        obs = submission.generate_sample_xml()
+        exp = ''.join([l.strip() for l in SAMPLEXML.splitlines()])
+        self.assertEqual(obs, exp)
 
-        e._generate_library_descriptor(elm, 'sample', 'libconsprot')
-        exp = ''.join([l.strip() for l in GENLIBDESC.splitlines()])
-        obs = ET.tostring(elm)
+    def test_generate_experiment_xml(self):
+        submission = EBISubmission(2, 'ADD')
+        # removing samples so test text is easier to read
+        keys_to_del = ['1.SKD6.640190', '1.SKM6.640187', '1.SKD9.640182',
```

Comment:
Are you always deleting the same keys? If so, you can either create that list on the setup or actually remove those keys already in the setup